src/Gd/Imagine.php

```diff
@@ -239,4 +249,19 @@ private function withoutExceptionHandlers($callback)
 
         return $result;
     }
+
+    private function isWebp(string $data)
+    {
+        return strncmp( substr( $data, 8, 7 ), "WEBPVP8", 7 ) === 0;
+    }
+
+    private function loadWebp(string $data)
+    {
+        $tmpfile = tempnam(sys_get_temp_dir(), 'imaginewebp_');
```

Comment:
I think that:
1. The temporary directory should be configurable
2. We should delete the temporary file once we've done reading it

Reply:
I'm not sure if that's necessery. Why would we want to configure the temp directory for just loading the file? 

Also if we want to delete the file we should probably do this somewhere else - we give the resource back and we cannot remove file that is currently opened as resources AFAIK. Perhaps on destructor of the class? To be honest - if it's in /tmp dir then filesystem will clear it eventually so it seemed like a "safe" choice :)

Reply:
I've seen crappy hosting providers where the directory returned from `sys_get_temp_dir()` was not writable...

About deleting the file: PHP loads the file content in memory, so the file can be safely deleted.

Reply:
Ok, will fix that in next commit after we resolve the second issue.

Reply:
Ok, will fix that in next commit after we resolve the second issue.